pkg/operator/controllers/checker/serviceprincipalchecker.go

```diff
@@ -79,7 +79,9 @@ func (r *ServicePrincipalChecker) Check(ctx context.Context) error {
 		updateFailedCondition(cond, err)
 	}
 
-	spDynamic, err := dynamic.NewValidator(r.log, &azEnv, resource.SubscriptionID, nil, dynamic.AuthorizerClusterServicePrincipal)
+	// TODO: Add a new limited interface (e.g ServicePrincipalValidator) and create a new constructor with less params
+	// ^ Separate PR
```

Comment:
Oh, I almost forgot about this one. I'll work on this shortly because having `nil`s for some args is getting more and more confusing.

Reply:
>having `nil`s for some args is getting more and more confusing.

Sorry, I know I can be pretty confusing when I argue.

Reply:
https://github.com/Azure/ARO-RP/pull/1615 addresses it